addons/web/static/tests/webclient/webclient_tests.js

```diff
@@ -74,3 +74,36 @@ QUnit.test("webclient for a non superuser", async (assert) => {
     const webClient = await mount(WebClient, { env, target });
     assert.doesNotHaveClass(webClient.el, "o_is_superuser");
 });
+
+QUnit.test("control-click propagation stopped on <a href/>", async (assert) => {
+    assert.expect(7);
+    class MyComponent extends Component {
+        /** @param {MouseEvent} ev */
+        onclick(ev) {
+            assert.step(ev.ctrlKey ? "ctrl-click" : "click");
+        }
+    }
+    MyComponent.template = xml`<a href="#" class="MyComponent" t-on-click="onclick" />`;
+    clearRegistryWithCleanup(mainComponentRegistry);
+    const target = getFixture();
+    let env = await makeTestEnv(baseConfig);
+
+    // Mount the component as standalone and control-click the <a href/>
+    const standaloneComponent = await mount(MyComponent, { env, target });
+    assert.verifySteps([]);
+    await triggerEvent(standaloneComponent.el, "", "click", { ctrlKey: false });
+    await triggerEvent(standaloneComponent.el, "", "click", { ctrlKey: true });
+    assert.verifySteps(["click", "ctrl-click"]);
+    standaloneComponent.destroy();
+
+    // Register the component as a main one, mount the webclient and control-click the <a href/>
+    clearRegistryWithCleanup(mainComponentRegistry);
+    mainComponentRegistry.add("mycomponent", { Component: MyComponent });
+    env = await makeTestEnv(baseConfig);
+    const webClient = await mount(WebClient, { env, target });
+    assert.verifySteps([]);
+    await triggerEvent(webClient.el, ".MyComponent", "click", { ctrlKey: false });
+    await triggerEvent(webClient.el, ".MyComponent", "click", { ctrlKey: true });
+    assert.verifySteps(["click"]);
```

Comment:
Should I assert that the event is non default prevented ?
How could I do that ?

Reply:
I could rewire the preventDefault method on each event
```js
  const clickEvent = new MouseEvent("click", { bubbles: true, cancelable: true, ctrlKey: true });
  clickEvent.preventDefault = ()=> assert.step("defaultPrevented");
  element.dispatchEvent(clickEvent);
```

Reply:
you could add another listener on the element you click on, and check that the event is prevent defaulted

Reply:
Here I want to assert it is NOT default prevented.
Also, I cannot add another listener as it will not get called (the webclient stops immediatly  the propagation on capturing phase)

Reply:
effectivement :-) ok then